lib/plugins/aws/package/compile/events/alb/lib/validate.js

```diff
@@ -89,21 +78,13 @@ module.exports = {
   },
 
   validateListenerArn(listenerArn, functionName) {
-    if (!listenerArn) {
-      throw new this.serverless.classes.Error(
-        `listenerArn is missing in function "${functionName}".`
-      );
-    }
     // If the ARN is a ref, use the logical ID instead of the ALB ID
     if (_.isObject(listenerArn)) {
       if (listenerArn.Ref) {
         return { albId: listenerArn.Ref, listenerId: listenerArn.Ref };
       }
-      throw new this.serverless.classes.Error(
-        `Invalid ALB listenerArn in function "${functionName}".`
-      );
     }
-    const matches = listenerArn.match(ALB_LISTENER_PATTERN);
+    const matches = listenerArn.match(new RegExp(this.ALB_LISTENER_PATTERN));
```

Comment:
It'll be nicer to define `this.ALB_LISTENER_PATTERN` as regExp instance, and for config schema, simply coerce it to string via `String(this.ALB_LISTENER_PATTERN)`

With above we create a new object instance for one time validation and it doesn't feel clean.

Reply:
even `.toString()` works

Reply:
Yes, it'll return regexp pattern as string (`String(regExp)` and `regExp.toString()` are equivalent)

Reply:
actually both `String(regExp)` and `regExp.toString()` output the enclosing forward slashes, but ajv expects patterns without them. `regExp.source` is what we want.

Reply:
Indeed, I forgot about that, yes `regExp.source` is the way to go